### PR TITLE
Fix NPE when no version name is set

### PR DESCRIPTION
**Pull request by mrober**
Fix NPE when no version name is set. This will make it easier for apps to have Android tests while including the Sessions SDK. Also removed references to ApplicationInfoFlags to make it easier to minify apps with older api levels.

When the version name is unset, use the version code.

### CODE REVIEW

**Comment by mrober** [comment]
#5195

File: firebase-sessions/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
 # Unreleased
 
+* [fixed] Fixed NPE when no version name is set.
```
Comment (visumickey) [inline firebase-sessions/CHANGELOG.md:3]: Can you add a reference to the Github issue here?

File: firebase-sessions/src/main/kotlin/com/google/firebase/sessions/SessionEvents.kt [hunk 2]
```diff
@@ -74,7 +75,7 @@ internal object SessionEvents {
       androidAppInfo =
         AndroidApplicationInfo(
           packageName = packageName,
-          versionName = packageInfo.versionName,
+          versionName = packageInfo.versionName ?: buildVersion,
```
Comment (visumickey) [inline firebase-sessions/src/main/kotlin/com/google/firebase/sessions/SessionEvents.kt:78]: Is build version guaranteed?

File: firebase-sessions/src/main/kotlin/com/google/firebase/sessions/settings/LocalOverrideSettings.kt [hunk 3]
```diff
@@ -18,32 +18,21 @@ package com.google.firebase.sessions.settings
...
 import android.content.Context
 import android.content.pm.PackageManager
-import android.os.Build
 import android.os.Bundle
 import kotlin.time.Duration
 import kotlin.time.DurationUnit
 import kotlin.time.toDuration
 
 internal class LocalOverrideSettings(context: Context) : SettingsProvider {
+  @Suppress("DEPRECATION") // TODO(mrober): Use ApplicationInfoFlags when target sdk set to 33
```
Comment (visumickey) [inline firebase-sessions/src/main/kotlin/com/google/firebase/sessions/settings/LocalOverrideSettings.kt:27]: With the version based checks removed, do we know if the new version APIs are backward compatible?

**Review by visumickey — APPROVED** [review]
Generally fine with the changes. But one thing to confirm before landing this PR. We need to make sure that the play store prevents submitting apps without `version_name`. As long as the impact is only with the tests scenario, this change looks good to me.

File: firebase-sessions/src/main/kotlin/com/google/firebase/sessions/SessionEvents.kt (hunk 2 above)
Comment (mrober) [reply firebase-sessions/src/main/kotlin/com/google/firebase/sessions/SessionEvents.kt:78]: The `buildVersion` is the `versionCode` as a String. When no versionCode is set, it returns 0. Play Store won't let apps upload two versions with the same versionCode. So it'll never be null, and it'll always be unique for apps published to the Play Store.

File: firebase-sessions/src/main/kotlin/com/google/firebase/sessions/settings/LocalOverrideSettings.kt (hunk 3 above)
Comment (mrober) [reply firebase-sessions/src/main/kotlin/com/google/firebase/sessions/settings/LocalOverrideSettings.kt:27]: They are, I tested it. And Firebase will be setting the target sdk to 33 soon anyway, then we can remove this check and use the new api only.

**Comment by mrober** [comment]
> Generally fine with the changes. But one thing to confirm before landing this PR. We need to make sure that the play store prevents submitting apps without `version_name`. As long as the impact is only with the tests scenario, this change looks good to me.

Yes you have to set both versionCode and versionName for Play Store https://developer.android.com/studio/publish/versioning#versioningsettings and the versionCode cannot be reused.

**Pull request merged**